marlowe-playground-client/src/BottomPanel/View.purs

```diff
@@ -0,0 +1,65 @@
+module BottomPanel.View (render) where
+
+import Prelude hiding (div)
+import BottomPanel.Types (Action(..), State, _panelView, _showBottomPanel)
+import Data.Lens (to, (^.))
+import Data.Maybe (Maybe(..))
+import Halogen.Classes (aHorizontal, accentBorderBottom, active, activeClass, closeDrawerArrowIcon, collapsed, flex, flexTen, footerPanelBg, minimizeIcon)
+import Halogen.HTML (ClassName(..), HTML, a, a_, div, img, li, section, text, ul)
+import Halogen.HTML.Events (onClick)
+import Halogen.HTML.Properties (alt, class_, classes, src)
+
+type PanelTitle panel
+  = { view :: panel
+    , classes :: Array ClassName
+    , title :: String
+    }
+
+render ::
+  forall p panel panelAction.
+  Eq panel =>
+  Array (PanelTitle panel) ->
+  (panel -> HTML p (Action panel panelAction)) ->
+  State panel ->
+  HTML p (Action panel panelAction)
+render panelTitles panelContents state =
```

Comment:
a comment with some explanation of the parameters (e.g. panelTitles is an ordered list of the tab titles)

Reply:
Added some comments